### ISSUE TITLE
qt 5.7 qt quick controls built in material style

### ISSUE THREAD

**Issue by swex** [papyros/qml-material#383]
Hi there just found out that in QT 5.7 material design will be available by default as built-in theme.
[qt 5.7 new features](https://wiki.qt.io/New_Features_in_Qt_5.7)
- Have you guys heard about that?
- What do you think about that?
- Mb better to go with mainline qml material?
##

**Comment by iBelieve** [comment]
Yes, we've heard about it. While I haven't looked into QtQuick.Controls 2 very much, or the material theme, my current understanding is that it's just a theme for the normal controls. You don't have any material-specific components like the bottom sheet, toolbar, navigation drawer, etc., and you don't have any material-specific properties like primary/accent colors or elevation.

So while I would love to use as much as we can from QtQuck.Controls 2.0 and the material theme, I think we'll still have to ship additional material-specific components on top of it to build apps that fully use Material Design.

**Comment by marco-piccolino** [comment]
the current qml-material implementation will definitely be beneficial for
Qt up to 5.6, which by the way is a LTS version.
For the future, I wonder if it's worth to start thinking about the
additional components in terms of general Controls 2 add-ons (maybe
implement the basic functionality in C++ as Controls 2 do) and provide the
material look as styles for them too. Also, some Components could extend
existing 2.0 Controls and provide additional styling and properties (e.g.
ActionButton, if not provided in Controls 2, could extend Button and
implement a specific style).

2016-02-01 18:06 GMT+01:00 Michael Spencer notifications@github.com:

**Comment by geiseri** [comment]
From my understanding @iBeliever is correct.  This is just a "style" and not actually the full behaviors for the UI.  This means no matter what you would still need QML material in some capacity.  If anything this would make the control's style unneeded.

**Comment by iBelieve** [comment]
So, I'm starting to look into a QtQuick.Controls 2 port of QML Material, and after learning a bit more about how Controls 2 works, my goal is (1) contribute as much as I can of QML Material into the Material Controls 2 theme itself (aka, improving the Controls 2 Material style), and (2) building a new small library of just additional components, not additional styles.

As of right now, I'm thinking of keeping the new library separate so this can continue to be developed for QtQuick.Controls 1 and Qt 5.6 as it is a LTS version. I'm also thinking about naming the new library Paper or QuickPaper so it's not confused with the built-in Material style/components in QtQuick.Controls 2.

So, at least for now, here's the new library: https://github.com/iBeliever/quickpaper. Anyone is welcome to contribute to that once I get an initial commit pushed up.

And my first contribution to the material style in Controls 2, adding a nice ElevationEffect component and proper elevation support: https://codereview.qt-project.org/#/c/157640/. Hopefully it gets merged 😄 

I'll be backporting the ElevationEffect into QML Material so we have better elevation support here as well.

**Comment by marco-piccolino** [comment]
Glad to hear this, it sounds like the right move to me.
Looking forward to help out!
Il 29 apr 2016 9:19 PM, "Michael Spencer" notifications@github.com ha
scritto:

So, I'm starting to look into a QtQuick.Controls 2 port of QML Material,
and after learning a bit more about how Controls 2 works, my goal is (1)
contribute as much as I can of QML Material into the Material Controls 2
theme itself (aka, improving the Controls 2 Material style), and (2)
building a new small library of just additional components, not additional
styles.

As of right now, I'm thinking of keeping the new library separate so this
can continue to be developed for QtQuick.Controls 1 and Qt 5.6 as it is a
LTS version. I'm also thinking about naming the new library Paper or
QuickPaper so it's not confused with the built-in Material style/components
in QtQuick.Controls 2.

So, at least for now, here's the new library:
https://github.com/iBeliever/quickpaper. Anyone is welcome to contribute to
that once I get an initial commit pushed up.

And my first contribution to the material style in Controls 2, adding a
nice ElevationEffect component and proper elevation support:
https://codereview.qt-project.org/#/c/157640/. Hopefully it gets merged 😄

I'll be backporting the ElevationEffect into QML Material so we have better
elevation support here as well.

**Comment by walkthetalk** [comment]
glad to hear that, the quick control 2 has improved performance/speed, especially on low-end device.
e.g. devices without gpu, only soft-render.
